source/dlangbot/github_api.d

```diff
@@ -378,18 +393,13 @@ struct GHIssue
         assert(issue.repoSlug == "dlang/phobos");
     }
 
-    // warning: not all fields can be filled
-    PullRequest toPullRequest() const
+    PullRequest pullRequest() const
     {
-        PullRequest pr;
-        pr.base.repo = PullRequest.Repo.init;
-        pr.base.repo.fullName = repoSlug;
-        static foreach (symbol; ["number", "state", "title", "user", "assignee",
-                    "createdAt", "updatedAt", "closedAt"])
-        {
-            mixin("pr." ~ symbol ~ " = " ~ symbol ~ ";");
-        }
-        return pr;
+        assert(isPullRequest);
+
+        return ghGetRequest(_pullRequest.url)
+                .readJson
+                .deserializeJson!PullRequest;
```

Comment:
This will _silently_ lead to a lot more requests.

Reply:
Hard to avoid, issues have no head commit, but on the plus side we're saving many requests because merge attempts for non-green PRs can be canceled early on.